symbol/scope.go

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 De-labtory
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package symbol
+
+// NewScope() makes a new scope.
+func NewScope() *Scope {
+	s := make(map[string]Symbol)
+	return &Scope{s, nil}
+}
+
+// NewEnclosedScope makes a new scope which has outer scope.
+func NewEnclosedScope(outer *Scope) *Scope {
+	scope := NewScope()
+	scope.outer = outer
+	return scope
+}
+
+// Scope represent a variable's scope.
+type Scope struct {
+	store map[string]Symbol
+	outer *Scope
```

Comment:
maybe we don't need `outer` for now. because our contract consists of functions, no nested function and block statement have same scope, i.e.
```
func foo() {
    // scope1

    if (expression) {
        // same scope with scope1
    }
}
```